docs/shared/connection_params.md

```diff
@@ -0,0 +1,45 @@
+### Optional Connection Parameters
```

Comment:
As mentioned earlier, this one to go within a separate PR.
Also not keen on `shared` name.